File: app/src/main/java/com/xdk78/nimebox/adapter/ArticleAdapter.java
package com.xdk78.nimebox.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.xdk78.nimebox.R;
import com.xdk78.nimebox.model.Article;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;


/**
 * Created by xdk78 on 2017-05-14.
 */

public class ArticleAdapter extends RecyclerView.Adapter<ArticleAdapter.ArticleiewHolder> {

    private List<Article> items = new ArrayList<>();
    private Context context;

    public ArticleAdapter(List<Article> items, Context context) {
        this.items = items;
        this.context = context;
    }

    public void addItems(List<Article> items) {
        this.items.addAll(items);
        notifyDataSetChanged();
    }

    @Override
    public ArticleiewHolder onCreateViewHolder(ViewGroup viewGroup, final int i) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.article_layout, viewGroup, false);
        return new ArticleiewHolder(view);
    }

    @Override
    public void onBindViewHolder(ArticleiewHolder viewHolder, final int i) {
        Article item = items.get(i);

        viewHolder.title.setText(item.getTitle());

        Glide.with(context)
                .asDrawable()
                .load(item.getImage().replace(" ", "%20"))
                .into((viewHolder).image);

        viewHolder.description.setText(item.getDescription());

    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class ArticleiewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.article_title)
        TextView title;

        @BindView(R.id.article_description)
        TextView description;

        @BindView(R.id.article_image)
        ImageView image;

        public ArticleiewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/java/com/xdk78/nimebox/adapter/ArticleAdapter.kt
package com.xdk78.nimebox.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.xdk78.nimebox.R
import com.xdk78.nimebox.mvp.model.ArticleModel
import org.jetbrains.anko.find

/**
 * Created by obsidiam on 11/3/17.
 */
class ArticleAdapter(private var context: Context, private var items: MutableList<ArticleModel>) : RecyclerView.Adapter<ArticleAdapter.ArticleAdapterViewHolder>() {

    override fun onBindViewHolder(holder: ArticleAdapterViewHolder, position: Int) {
        val items = items[position]
        holder.title.text = items.title

        Glide.with(context)
                .load(items.image)
                .into(holder.image)

        holder.description.text = items.description
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ArticleAdapterViewHolder {
        val layoutInflater = LayoutInflater.from(parent?.context)
        return ArticleAdapterViewHolder(layoutInflater.inflate(R.layout.article_item, parent, false))
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class ArticleAdapterViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val title: TextView = view.find(R.id.article_title)
        val description: TextView = view.find(R.id.article_description)
        val image: ImageView = view.find(R.id.article_image)
    }
}